substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVTypeRecord.java

```diff
@@ -117,10 +159,307 @@ private static int alignPadded4(byte[] buffer, int originalpos) {
         return pos;
     }
 
+    static final class CVTypePrimitive extends CVTypeRecord {
+
+        private final int length;
+
+        CVTypePrimitive(short cvtype, int length) {
+            super(cvtype);
+            assert cvtype < FIRST_TYPE_INDEX;
+            this.length = length;
+            setSequenceNumber(cvtype);
+        }
+
+        @Override
+        public int computeSize(int initialPos) {
+            GraalError.shouldNotReachHere();
+            return 0;
+        }
+
+        @Override
+        protected int computeContents(byte[] buffer, int initialPos) {
+            GraalError.shouldNotReachHere();
+            return 0;
+        }
+
+        @Override
+        public int hashCode() {
+            GraalError.shouldNotReachHere();
+            return 0;
+        }
+
```

Comment:
These should not reach here's smell a bit odd -- as if you ought to have an intermediate abstract subclass below CVTypeRecord for the all the subclasses that have content and hence implement these methods. That woudl allow you to rely on the type system to ensure that these methods don't get called.

Is there a reason why you cannot structure the class hierarchy like that? or is it just a question of the difficulty you would face retrofitting it into the original design?

Reply:
It's unclear what changes would be required for aarch64 debug info, so I thought I'd catch this one here, just to ensure it's not assumed to work.  I'm happy to remove the test altogether, but I'd rather not restructure the code until/unless a Windows aarch64 port is underway.

Reply:
Ok, so this is just future proofing That's fine.